toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -2159,6 +2171,56 @@ func runPostInstallScripts(installChroot *safechroot.Chroot, config configuratio
 	return
 }
 
+func cleanupTdnfCache(installChroot *safechroot.Chroot) error {
+	const (
+		squashErrors      = false
+		rpmCacheDirectory = "/var/cache/tdnf"
+	)
+
+	ReportActionf("Cleaning tdnf cache")
+	err := installChroot.UnsafeRun(func() error {
+		logger.Log.Infof("Cleaning tdnf cache")
+		chrootErr := shell.ExecuteLive(squashErrors, "tdnf", "clean", "all")
+		return chrootErr
+	})
+
+	if err != nil {
+		err = fmt.Errorf("failed to cleanup tdnf cache:\n%w", err)
+		return err
+	}
+
+	// Remove all files and subdirectories in the tdnf cache directory, but leave
+	// the directory since it's owned by the tdnf package.
+	cacheDir := filepath.Join(installChroot.RootDir(), rpmCacheDirectory)
+	err = filepath.WalkDir(cacheDir, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		// Ignore the cache directory itself
+		if path == cacheDir {
+			return nil
+		}
+		logger.Log.Debugf("Removing tdnf cache file (%s)", path)
+		removeErr := os.RemoveAll(path)
+		if removeErr != nil {
+			return fmt.Errorf("failed to remove tdnf cache file (%s):\n%w", path, removeErr)
+		}
+		// Do not recurse into directories, walk gets confused if we remove directories
+		// while walking them. os.RemoveAll will recursively remove the directory so no
+		// need to do it via walk.
+		if d.IsDir() {
+			return fs.SkipDir
+		} else {
+			return nil
+		}
+	})
+	if err != nil {
+		err = fmt.Errorf("failed to scan tdnf cache directory for files to remove(%s):\n%w", cacheDir, err)
+		return err
+	}
```

Comment:
Can we move this into `file.go`? Seems like a useful function.

Also, we could do something similar to (or identical) `RemoveContents` from [here](https://stackoverflow.com/a/33451503/11313995). It doesn't require separate logic for skipping the top directory and subdirectories.

Reply:
Done